### PR TITLE
Remove yet more type-ignores in dynamo/inductor

### PR DESCRIPTION
**Pull request by int3**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114684

Probably the last big batch for a while

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -1536,9 +1536,11 @@ def _mask_value(value, default):
                 masked_value = _mask_value(value, default)
 
             if reduction_type in {"argmax", "argmin"}:
-                accumulator_index = self.cse.generate(
-                    self.compute,
-                    f"tl.broadcast_to({reduction_range_prefix}index, {masked_value}.shape)",
+                accumulator_index = str(
```
Comment (int3) [inline torch/_inductor/codegen/triton.py:1539]: Previously, `accumulator_index` had type CSEVariable here but type `str` later on, leading mypy to raise an error. However, `accumulator_index` is only used in a string context throughout this method, so eagerly stringify-ing it (so it is a `str` type throughout) should be safe.

File: torch/_inductor/select_algorithm.py [hunk 2]
```diff
@@ -386,7 +385,7 @@ def call_kernel(self, name: str, node: Optional[ir.IRNode] = None):
             ] + [meta]
             grid_call = f"{self.grid_fn.__module__}.{self.grid_fn.__name__}({', '.join(grid_call)})"
             wrapper.writeline(
-                f"{name}.run({call_args}, grid={grid_call}, stream={stream_name})"
+                f"{name}.run({', '.join(call_args)}, grid={grid_call}, stream={stream_name})"
```
Comment (int3) [inline torch/_inductor/select_algorithm.py:388]: avoids type redefinition

File: torch/_inductor/triton_heuristics.py [hunk 3]
```diff
@@ -89,15 +89,15 @@ def autotune_hints_to_configs(
...
     configs = []
 
     for hint in hints:
         if hint == AutotuneHint.ELEMENTS_PER_WARP_32:
             if len(size_hints) == 1:
-                xyz_options = ((block_size // 4,),)
+                xyz_options = ((block_size // 4, None, None),)
             elif len(size_hints) == 2:
-                xyz_options = ((block_size // 4, 1), (1, block_size // 4))
+                xyz_options = ((block_size // 4, 1, None), (1, block_size // 4, None))
```
Comment (int3) [inline torch/_inductor/triton_heuristics.py:100]: Using a variable number of tuple sizes causes `mypy` to be confused when typechecking the call to `triton_config` below; it assumes that the tuple can have any number of elements, and so considers `num_elements_per_warp` to be defined twice. This change makes it clear that `xyz_options` can only have 3 elements. The `y` and `z` parameters to `triton_config` have default arguments of `None` anyway, so this is safe.

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**